File: Programmers/kakao/신규아이디추천.py
import re

def solution(new_id):
    #1단계
    answer = new_id.lower()
    #2단계
    answer = re.sub('[~!@#$%^&*\(\)=+\[\{\]\}:?,\<\>/]', '', answer)

    #3단계
    answer = re.sub('\.+', '.', answer)

    #4단계
    if answer:
        if answer[0] == '.':
            answer = answer[1:]

    if answer:
        if answer[-1] == '.':
            answer = answer[:-1]

    #5단계
    if not answer:
        answer = 'a'

    #6단계
    answer = answer[:15]
    if answer:
        if answer[-1] == '.':
            answer = answer[:-1]

    #7단계
    if len(answer) < 3:
        answer = answer + answer[-1]*(3 - len(answer))

    return answer


solution("z-+.^.")

File: Java/src/com/company/level_1/신규아이디추천.java
package com.company.level_1;

public class 신규아이디추천 {

    public String solution(String new_id) {
        String answer = "";

        //case1
        new_id = new_id.toLowerCase();

        //case2
        new_id = new_id.replaceAll("[^a-z0-9-_.]", "");

        //case3
        new_id = new_id.replaceAll("[.]+",".");
        //new_id = new_id.replaceAll("[.]{2,}", ".");

        //case4
        StringBuilder stringBuilder = new StringBuilder(new_id);
        if (stringBuilder.length() > 0) {
            if(stringBuilder.charAt(0) == '.') stringBuilder.deleteCharAt(0);
        }
        if (stringBuilder.length() > 0) {
            if(stringBuilder.charAt(stringBuilder.length()-1) == '.') stringBuilder.deleteCharAt(stringBuilder.length()-1);
        }

        //case5
        if (stringBuilder.length() < 1) stringBuilder.append("a");

        //case6
        if (stringBuilder.length() > 15) {
            stringBuilder.delete(15, stringBuilder.length());
            if (stringBuilder.charAt(stringBuilder.length()-1) == '.'){
                stringBuilder.deleteCharAt(stringBuilder.length() - 1);
            }
        }


        //case7
        if (stringBuilder.length() < 3) {
            while (stringBuilder.length() < 3){
                stringBuilder.append(stringBuilder.charAt(stringBuilder.length()-1));
            }
        }

        answer = new String(stringBuilder);

        return answer;
    }

}
